core/src/main/java/org/apache/iceberg/HistoryTable.java

```diff
@@ -82,6 +82,11 @@ private DataTask task(TableScan scan) {
       // override planFiles to avoid the check for a current snapshot because this metadata table is for all snapshots
       return CloseableIterable.withNoopClose(HistoryTable.this.task(this));
     }
+
+    @Override
+    protected String tableType() {
+      return String.valueOf(HistoryTable.this.metadataTableType());
```

Comment:
For scans that inherit from `StaticTableScan`, it's possible to get the table type from the surrounding class's method.

For the entries that extend `BaseAllMetadataTableScan`, it's not possible to call methods from the surrounding class so I used the metadata table type enum values directly (such as [here](https://github.com/apache/iceberg/pull/2617/files#diff-c106422c51344f58d260f5578bfc8e720106dcd441f8348660155bb0b331062bR87-R91)).

Not really in love with one style or another, though this one here would be harder to mess up in a refactor as it's more DRY (at the expense of being a bit more verbose).